File: frontend/src/types/document.ts
export interface DocumentChunk {
  id: string;
  document_id: string;
  section_name: string;
  text: string;
  chunk_index: number;
}

export interface Document {
  id: string;
  name: string;
  upload_date: string;
  status: "processing" | "ready" | "error";
  chunk_count: number;
  raw_text: string;
}

export interface DocumentDetail extends Document {
  chunks: DocumentChunk[];
}

export interface DocumentListResponse {
  documents: Document[];
}

File: backend/models/document.py
from datetime import datetime

from pydantic import BaseModel


class DocumentChunk(BaseModel):
    id: str
    document_id: str
    section_name: str
    text: str
    chunk_index: int


class Document(BaseModel):
    id: str
    name: str
    upload_date: datetime
    status: str  # "processing", "ready", "error"
    chunk_count: int
    raw_text: str


class DocumentDetail(Document):
    chunks: list[DocumentChunk]


class DocumentListResponse(BaseModel):
    documents: list[Document]
